apollo-federation/src/sources/connect/json_selection/lit_expr.rs

```diff
@@ -0,0 +1,522 @@
+// A LitExpr (short for LiteralExpression) is similar to a JSON value (or
+// serde_json::Value), with the addition of PathSelection as a possible leaf
+// value, so literal expressions passed to -> methods (via MethodArgs) can
+// capture both field and argument values and sub-paths, in addition to constant
+// JSON structures and values.
```

Comment:
Use `//!` for module documentation.

Reply:
https://github.com/apollographql/router/pull/5762/commits/8c18a5f99ccd094cd25dc0aaa3176fefcff26cc8